spec/controllers/admin_controller_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe AdminController, type: :controller do
+  controller do
+    def authenticate_admin_mock
+      authenticate_admin
+    end
+  end
+
+  describe '#authenticate_admin' do
+    subject { controller.authenticate_admin_mock }
```

Comment:
Since this is a controller spec, would it make more sense to treat the subject as a request? 

`get authenticate_admin` or something.